pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -200,12 +198,13 @@ func (l *podNICImpl) PlugPhase1(vmi *v1.VirtualMachineInstance, iface *v1.Interf
 			return err
 		}
 
-		if err := bindMechanism.preparePodNetworkInterfaces(); err != nil {
+		vmIface, err := bindMechanism.preparePodNetworkInterface()
+		if err != nil {
 			log.Log.Reason(err).Error("failed to prepare pod networking")
 			return createCriticalNetworkError(err)
 		}
 
-		if err := bindMechanism.setCachedInterface(); err != nil {
+		if err := l.setCachedInterface(&vmIface); err != nil {
```

Comment:
As said, vmIface is not going to be nil never, let's change `setCachedInterface` so it does not need a pointer.

Reply:
You can ignore this comment I was looking at wrong code, we just pass the pointer to the cacheFactory.